language/vm/vm-runtime/src/system_txn/block_metadata_processor.rs

```diff
@@ -0,0 +1,48 @@
+use crate::{
+    code_cache::module_cache::ModuleCache,
+    data_cache::BlockDataCache,
+    process_txn::execute::ExecutedTransaction,
+    txn_executor::{TransactionExecutor, LIBRA_SYSTEM_MODULE},
+};
+use lazy_static::lazy_static;
+use libra_types::{
+    block_metadata::BlockMetadata,
+    identifier::Identifier,
+    transaction::TransactionOutput,
+    vm_error::{StatusCode, VMStatus},
+};
+use vm::transaction_metadata::TransactionMetadata;
+use vm_runtime_types::value::Value;
+
+lazy_static! {
+    static ref BLOCK_PROLOGUE: Identifier = Identifier::new("block_prologue").unwrap();
+}
+
+pub(crate) fn process_block_metadata<'alloc, P>(
+    block_metadata: BlockMetadata,
+    module_cache: P,
+    data_cache: &mut BlockDataCache<'_>,
+) -> TransactionOutput
+where
+    P: ModuleCache<'alloc>,
+{
+    // TODO: How should we setup the metadata here?
```

Comment:
I think we should sort this out before landing this code; the semantics of this is something you could easily imagine VM implementations doing differently. The logic should almost certainly be here and not hidden somewhere like` Default`.

A couple of ideas:
- Change `TransactionMetadata` to a trait and use an implementation that `panic`s when you try to read anything. This ensures that no one can observe these nonsensical values.
- In the absence of that, I think the most important consideration is figuring out the sender address. Having a notion of a "null address" (probably 0x0...0) that is prohibited from containing modules or resources (i.e., `Account.create()` would explicitly rule out creating an account at the null address) is a concept that would be useful to us in several other places. If we have such a concept, that would be a natural choice for the sender address.
- I'm guessing the only value that actually gets read from the transaction metadata is `max_gas_amount` (to initialize the interpreter's gas counter). U64_MAX seems like the right choice here.